openmdao/drivers/genetic_algorithm_driver.py

```diff
@@ -389,7 +389,12 @@ def objective_callback(self, x, icase):
         nr_objectives = len(objs)
 
         # Single objective, if there is only one objective, which has only one element
-        is_single_objective = (nr_objectives == 1) and (len(objs) == 1)
+        if nr_objectives > 1:
+            is_single_objective = False
+        else:
+            for obj in objs.items():
+                break
+            is_single_objective = len(obj) == 1
```

Comment:
If objs is empty then obj will be undefined here.  I would move this line inside the for loop before the break.  Do we have a test for when no objectives are defined?

Reply:
Yes,, This driver calls `self._check_for_missing_objective()` in the run method.

Reply:
Ok.  I would still recommend moving that line into the for loop though.  It would make it more obvious that the code is correct.

Reply:
ok, done